apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -127,21 +137,37 @@ defmodule Aecore.Miner.Worker do
     latest_block_hash = BlockValidation.block_header_hash(latest_block.header)
     chain_state = Chain.chain_state(latest_block_hash)
 
-    txs_list = Map.values(Pool.get_pool())
-    ordered_txs_list = Enum.sort(txs_list, fn(tx1, tx2) -> tx1.data.nonce < tx2.data.nonce end)
+    blocks_for_difficulty_validation = if(latest_block.header.height == 0) do
+      Chain.get_blocks(latest_block_hash, Difficulty.get_number_of_blocks())
+    else
+      [_ | something] = Chain.get_blocks(latest_block_hash, Difficulty.get_number_of_blocks() + 1)
+      something
+    end
```

Comment:
I trust this logic is correct. *it looks weird*

Reply:
@nedelcho-delchev-tues maybe we can optimize this somehow to be better readable and add some comment

Reply:
I pushed a change directly to master, you can see if it is better and more understandable.
@cytadela8